guides/softlayer.md

```diff
@@ -0,0 +1,143 @@
+
+For example, the location id is used to specify the "region" provider where to spin your VM: 
+{% highlight java %}
+TemplateBuilder templateBuilder = context.getComputeService().templateBuilder();
+templateBuilder.locationId("ams01");
+{% endhighlight %}
+
+If you need to specify a public image supported by SoftLayer can be referenced as follows:
+{% highlight java %}
+templateBuilder.imageId("CENTOS_6_64");
+{% endhighlight %}
+
+If you want instead to specify a private image, you need to set the globalIdentifier of a VirtualGuestBlockDeviceTemplateGroup
+{% highlight java %}
+templateBuilder.imageId("3d7697d8-beef-437a-8921-5a2a18bc116f");
+{% endhighlight %}
+
+Notice, if you don't know the globalIdentifier upfront you can use the following curl command
+
+curl -uusername:apiKey https://api.softlayer.com/rest/v3/SoftLayer_Account/getBlockDeviceTemplateGroups?objectMask=children.blockDevices.diskImage.softwareReferences.softwareDescription
+
+and use the globalIdentifier desired
+
+jclouds is able to leverage SoftLayer CCI specific-options described at [createObject](http://sldn.softlayer.com/reference/services/SoftLayer_Virtual_Guest/createObject) by doing the following:
+{% highlight java %}
+SoftLayerTemplateOptions options = template.getOptions().as(SoftLayerTemplateOptions.class);
+// domain
+options.domainName("live.org");
+// multi-disk
+options.blockDevices(ImmutableList.of(25, 400, 400));
+// disk type (SAN, LOCAL)
+options.diskType("SAN");
+//tags
+options.tags(ImmutableList.of("jclouds"));
+// primaryNetworkComponent.networkVlan.id
+options.primaryNetworkComponentNetworkVlanId(vlanId);
+// primaryBackendNetworkComponent.networkVlan.id
+options.primaryBackendNetworkComponentNetworkVlanId(backendVlanId);
+{% endhighlight %}
+NB: notice, vlandId and backendVlanId should be the internal IDs and not the vlan number! One can easily retrieve it from the SoftLayer web UI by clicking on VLAN number link of `https://control.softlayer.com/network/vlans`
+The url will show something like `https://control.softlayer.com/network/vlans/1234567` where the vlanId is *1234567*
+
+### Provider-specific APIs
+When you need access to SoftLayer features, use the provider-specific context
+{% highlight java %}
+SoftLayerApi api = computeServiceContext.unwrapApi(SoftLayerApi.class);
+// and then get access to the delegate API groups
+api.getVirtualGuestApi();
+api.getDatacenterApi();
+api.getSoftwareDescriptionApi();
+api.getVirtualGuestBlockDeviceTemplateGroupApi();
+api.getAccountApi();
+
+// Be sure to close the context when done
+computeServiceContext.close();
+{% endhighlight %}
+
+## SoftLayer Object Storage
+SoftLayer Object Storage is an OpenStack® based blobstore storage system.
+
+{% highlight java %}
+// Get a context with softlayer that offers the portable BlobStore API
+BlobStoreContext context = ContextBuilder.newBuilder("swift")
```

Comment:
From looking at [Auth Endpoints](http://sldn.softlayer.com/article/Introduction-Object-Storage#Authentication_Endpoints) it seems clear it's TempAuth only. 

These instructions really shouldn't be written based on deprecated code. We really don't need another thing blocking us from removing `"swift"`. 

The proper thing to do here is to update the instructions to use the [SwiftAuthenticationModule](http://jclouds-javadocs.elasticbeanstalk.com/org/jclouds/openstack/swift/v1/config/SwiftAuthenticationModule.html)

    When KeystoneProperties.CREDENTIAL_TYPE is set to tempAuthCredentials, do not use Keystone. Instead, bridge TempAuth to Keystone by faking a service catalog out of the storage url. The endpoint must be set to the TempAuth url, usually ending in auth/v1.0/.

Unfortunately that will only be in the next release.

I propose we remove Object Storage from this particular PR and update this guide with the SwiftAuthenticationModule when the next version of jclouds is released.

The alternative is to leave it as is (basically recommending users to use deprecated code) and update this guide with the SwiftAuthenticationModule when the next version of jclouds is released.

Your call @andreaturli